hc2/headers/types/code_object_bundle.hpp

```diff
@@ -142,9 +142,8 @@ namespace hc2
             triple.cend(),
             [](char x) { return std::isdigit(x); });
         if (std::equal(triple.cbegin(), it, gfxip)) {
```

Comment:
@t-tye:
```
inline
hsa_isa_t triple_to_hsa_isa(const std::string& triple)
{
    static constexpr const char oldPrefix[] = "hcc-amdgcn--amdhsa-gfx";
    static constexpr const char newPrefix[] = "amdgcn-amd-amdhsa--gfx";

    std::string validatedTriple = triple;
    if ((triple.size() >= sizeof(oldPrefix) - 1) && std::equal(oldPrefix, oldPrefix + sizeof(oldPrefix) - 1, triple.c_str())) {
        // Support backwards compatibility with old naming.
        validatedTriple = newPrefix + triple.substr(sizeof(oldPrefix) - 1);
    }

    hsa_isa_t Isa = {0};
    if (HSA_STATUS_SUCCESS != hsa_isa_from_name(validatedTriple.c_str(), &Isa)) {
        Isa.handle = 0;
    }

    return Isa;
}
```

Reply:
I will add this and test it on all HCC/HIP and HIP samples.

Reply:
Here is an updated version that allows the offload kind to be the first element.

```
inline
hsa_isa_t triple_to_hsa_isa(const std::string& triple)
{
    static constexpr const char OffloadKind[] = "hcc-";
    hsa_isa_t Isa = {0};
    if ((triple.size() < sizeof(OffloadKind) - 1) || !std::equal(triple.c_str(),  triple.c_str() + sizeof(OffloadKind) - 1, triple.c_str()))) {
       return Isa;
    }
    std::string validatedTriple = triple.substr(sizeof(OffloadKind) - 1);
    static constexpr const char oldPrefix[] = "amdgcn--amdhsa-gfx";
    static constexpr const char newPrefix[] = "amdgcn-amd-amdhsa--gfx";

    if ((triple.size() >= sizeof(oldPrefix) - 1) && std::equal(oldPrefix, oldPrefix + sizeof(oldPrefix) - 1, triple.c_str())) {
        // Support backwards compatibility with old naming.
        validatedTriple = newPrefix + triple.substr(sizeof(oldPrefix) - 1);
    }

    if (HSA_STATUS_SUCCESS != hsa_isa_from_name(validatedTriple.c_str(), &Isa)) {
        Isa.handle = 0;
    }

    return Isa;
}
```

Reply:
Here is an updated version with correction from @scchan.

```
inline
hsa_isa_t triple_to_hsa_isa(const std::string& triple)
{
    static constexpr const char OffloadKind[] = "hcc-";
    hsa_isa_t Isa = {0};
    if ((triple.size() < sizeof(OffloadKind) - 1) || !std::equal(triple.c_str(),  triple.c_str() + sizeof(OffloadKind) - 1, triple.c_str()))) {
       return Isa;
    }
    std::string validatedTriple = triple.substr(sizeof(OffloadKind) - 1);
    static constexpr const char oldPrefix[] = "amdgcn--amdhsa-gfx";
    static constexpr const char newPrefix[] = "amdgcn-amd-amdhsa--gfx";

    if ((validatedTriple.size() >= sizeof(oldPrefix) - 1) && std::equal(oldPrefix, oldPrefix + sizeof(oldPrefix) - 1, validatedTriple.c_str())) {
        // Support backwards compatibility with old naming.
        validatedTriple = newPrefix + validatedTriple.substr(sizeof(oldPrefix) - 1);
    }

    if (HSA_STATUS_SUCCESS != hsa_isa_from_name(validatedTriple.c_str(), &Isa)) {
        Isa.handle = 0;
    }

    return Isa;
}
```